lib/galaxy/managers/histories.py

```diff
@@ -221,6 +232,16 @@ def serve_ready_history_export(self, trans, jeha):
         archive = trans.app.object_store.get_filename(jeha.dataset)
         return open(archive, mode='rb')
 
+    def serve_ready_history_export(self, trans, jeha):
+        """
+        Serves the history export archive for use as a streaming response so the file
```

Comment:
The file is not loaded into memory with the legacy method either, and the file handle is passed on to nginx completely freeing the handler. I think that is preferable over a streaming response, even if it is async. Is there a downside I am not thinking of with what is now the `legacy_serve_ready_history_export` method that would prevent using it?

Reply:
That's a good question :)
I just read the documentation on how to deal with files in FastAPI and thought that [StreamingResponse](https://fastapi.tiangolo.com/advanced/custom-response/#streamingresponse) was a better fit than [FileResponse](https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse) because we already had a file-like handle in `serve_ready_history_export`, but honestly, I have no idea which option is best as both seems to be streaming the file asynchronously in the end.

This has just reminded me of https://github.com/galaxyproject/galaxy/pull/11167#discussion_r559535763 which I promised to do :)

Do you think the way to go here is using a FileResponse but introducing the "send file middleware" to set the headers? Or what do you think should be the best approach?

Reply:
> Do you think the way to go here is using a FileResponse but introducing the "send file middleware" to set the headers?

Yeah, I think that sounds good to me. If you set the middleware here you also have access to the config: https://github.com/mvdbeek/galaxy/blob/611fb8fe8b3a1994e97d7399a6aa8202bffe7a77/lib/galaxy/webapps/galaxy/fast_app.py#L61

Reply:
Cool! Thanks! I will try!

Reply:
Done!, I hope this will work https://github.com/galaxyproject/galaxy/pull/12441, don't really know how to properly test it though.

Reply:
Alright, then I think what's left to do is to use the FileResponse ?

Reply:
Done in https://github.com/galaxyproject/galaxy/pull/12386/commits/29db23928a5e3937575c92e24922fc91dd930186 :)